rfq/manager.go

```diff
@@ -1054,6 +1054,190 @@ func (m *Manager) ChannelMatchesFully(ctx context.Context,
 	return true, nil
 }
 
+// ChannelWithSpecifier is a helper struct that combines the information of an
+// asset specifier that is satisfied by a channel with the channels' general
+// information.
+type ChannelWithSpecifier struct {
+	// Specifier is the asset Specifier that is satisfied by this channels'
+	// assets.
+	Specifier asset.Specifier
+
+	// ChannelInfo is the information about the channel the asset is
+	// committed to.
+	ChannelInfo lndclient.ChannelInfo
+
+	// AssetInfo contains the asset related info of the channel.
+	AssetInfo rfqmsg.JsonAssetChannel
+}
+
+// ComputeChannelAssetBalance computes the total local and remote balance for
+// each asset channel that matches the provided asset specifier.
+func (m *Manager) ComputeCompatibleChannelAssetBalance(ctx context.Context,
+	activeChannels []lndclient.ChannelInfo,
+	specifier asset.Specifier) ([]ChannelWithSpecifier, bool, error) {
+
+	var (
+		channels                 = make([]ChannelWithSpecifier, 0)
+		haveGroupedAssetChannels bool
+	)
+	for chanIdx := range activeChannels {
+		openChan := activeChannels[chanIdx]
+		if len(openChan.CustomChannelData) == 0 {
+			continue
+		}
+
+		var assetData rfqmsg.JsonAssetChannel
+		err := json.Unmarshal(openChan.CustomChannelData, &assetData)
+		if err != nil {
+			return nil, false, fmt.Errorf("unable to unmarshal "+
+				"asset data: %w", err)
+		}
+
+		if len(assetData.GroupKey) > 0 {
+			haveGroupedAssetChannels = true
+		}
+
+		// Check if the assets of this channel match the provided
+		// specifier.
+		pass, err := m.ChannelMatchesFully(
+			ctx, assetData, specifier,
+		)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if pass {
+			channels = append(channels, ChannelWithSpecifier{
+				Specifier:   specifier,
+				ChannelInfo: openChan,
+				AssetInfo:   assetData,
+			})
+		}
+	}
+
+	return channels, haveGroupedAssetChannels, nil
+}
+
+// chanIntention defines the intention of calling rfqChannel. This helps with
+// returning the channel that is most suitable for what we want to do.
+type chanIntention uint8
```

Comment:
We should export this since `RfqChannel` is exported. Also, `RfqChannel` method name is not exact in doc comment. Note also my comment on naming `RfqChannel`.